File: Leetcode/LL/LLCycle.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None
    
class Solution:
    def hasCycle(self, head):
        fast, slow = head, head
        while fast and fast.next:
            fast, slow = fast.next.next, slow.next
            if fast is slow:
                return True
        return False

if __name__=="__main__":
    head = ListNode(1)
    head.next = ListNode(2)
    head.next.next = ListNode(3)
    head.next.next.next = head
    print Solution().hasCycle(head)

File: Leetcode/LL/LLCycle.js
class ListNode {
    constructor(x) {
        this.val = x;
        this.next = null; 
    }
}

hasCycle = (head) => {
    let fast = head;
    let slow = head;
    while(fast && fast.next) {
        fast = fast.next.next;
        slow = slow.next;
        if(fast == slow) {
            return true;
        }
    }
    return false;
}

let head = new ListNode(2)
head.next = new ListNode(3)
head.next.next = new ListNode(4)
head.next.next.next = head
console.log(hasCycle(head))